### PR TITLE
add helm chart for capa infrastracture controller

### PR DESCRIPTION
**Pull request by anvddriesch**
Towards https://github.com/giantswarm/giantswarm/issues/15992

### CODE REVIEW

File: helm/cluster-api-provider-aws/templates/cluster-template-eks-managedmachinepool-vpccni.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
```
Comment (MarcelMue) [inline helm/cluster-api-provider-aws/templates/cluster-template-eks-managedmachinepool-vpccni.yaml:1]: I don't think we need the CRDs to create clusters in here

File: helm/cluster-api-provider-aws/Chart.yaml [hunk 2]
```diff
@@ -1,10 +1,6 @@
...
-icon: https://some.domain.com/logo.png
+apiVersion: "v1"
 name: cluster-api-provider-aws
-sources:
-- https://github.com/some-org/some-repo
-version: [[ .Version ]]
+description: "Helm chart for cluster-api-provider-aws."
+home: "https://github.com/giantswarm/cluster-api-provider-aws-app"
+version: "[[ .Version ]]"
+appVersion: "[[ .AppVersion ]]"
```
Comment (QuentinBisson) [inline helm/cluster-api-provider-aws/Chart.yaml:6]: I think you are missing a line break here

File: helm/cluster-api-provider-aws/templates/_helpers.tpl [hunk 3]
```diff
@@ -20,20 +10,16 @@ Common labels
...
 
 {{/*
 Selector labels
 */}}
 {{- define "labels.selector" -}}
-app.kubernetes.io/name: {{ include "name" . | quote }}
-app.kubernetes.io/instance: {{ .Release.Name | quote }}
-{{- end -}}
+app: {{ .Values.name | quote }}
+{{- end -}}
```
Comment (QuentinBisson) [inline helm/cluster-api-provider-aws/templates/_helpers.tpl:25]: Same line break here

File: helm/cluster-api-provider-aws/templates/_resource.tpl [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+helm.sh/chart: {{ include "chart" . | quote }}
+{{- end -}}
+
+{{/*
+Selector labels
+*/}}
+{{- define "labels.selector" -}}
+app: {{ .Values.name | quote }}
+{{- end -}}
```
Comment (QuentinBisson) [inline helm/cluster-api-provider-aws/templates/_resource.tpl:25]: Same line break

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+  labels:
+    cluster.x-k8s.io/provider: bootstrap-eks
+    control-plane: controller-manager
+  name: capa-eks-bootstrap-system
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations:
+    ${AWS_CONTROLLER_IAM_ROLE/#arn/eks.amazonaws.com/role-arn: arn}
```
Comment (MarcelMue) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:13]: Not sure if we have this IAM role. Can we remove this annotation? not sure how we do this in our aws-operator

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.5.0
+        imagePullPolicy: IfNotPresent
+        name: kube-rbac-proxy
+        ports:
+        - containerPort: 8443
+          name: https
+      - args:
+        - --metrics-addr=127.0.0.1:8080
+        - --enable-leader-election
+        - --feature-gates=MachinePool=${EXP_MACHINE_POOL:=false}
```
Comment (MarcelMue) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:251]:
```suggestion
        - --feature-gates=MachinePool=false
```

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+        command:
+        - /manager
+        image: gcr.io/k8s-staging-cluster-api-aws/eks-bootstrap-controller:v0.6.4
+        imagePullPolicy: Always
+        name: manager
+      metadata:
+        annotations:
+          iam.amazonaws.com/role: ${AWS_CONTROLLER_IAM_ROLE:=""}
```
Comment (MarcelMue) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:259]: Same as the annotation above

File: helm/cluster-api-provider-aws/templates/metadata.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# maps release series of major.minor to cluster-api contract version
+# the contract version may change between minor or major versions, but *not*
+# between patch versions.
+#
+# update this file only when a new major or minor version is released
+apiVersion: clusterctl.cluster.x-k8s.io/v1alpha3
```
Comment (MarcelMue) [inline helm/cluster-api-provider-aws/templates/metadata.yaml:6]: Don't think this is needed.

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+    control-plane: controller-manager
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: bootstrap-eks
+    control-plane: controller-manager
+  name: capa-eks-bootstrap-controller-manager
+  namespace: capa-eks-bootstrap-system
```
Comment (MarcelMue) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:223]: namespace should be giantswarm IMO

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 10]
```diff
@@ -0,0 +1,266 @@
...
+    cluster.x-k8s.io/provider: bootstrap-eks
+    control-plane: controller-manager
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: bootstrap-eks
+    control-plane: controller-manager
+  name: capa-eks-bootstrap-controller-manager
```
Comment (MarcelMue) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:222]:
```suggestion
  name: {{ include "resource.default.name" . }}
```
Naming can be used like this - it works a bit easier.

Comment (MarcelMue) [reply helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:222]: Easiest to check my PR maybe? Uncertain

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 11]
```diff
@@ -0,0 +1,266 @@
...
+  name: capa-eks-bootstrap-leader-election-rolebinding
+  namespace: capa-eks-bootstrap-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: capa-eks-bootstrap-leader-election-role
+subjects:
+- kind: ServiceAccount
+  name: capa-eks-bootstrap-controller-manager
+  namespace: capa-eks-bootstrap-system
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:167]:
```suggestion
  name: {{ include "resource.default.name" . }}
  namespace: {{ include "resource.default.namespace" . }}
```

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 12]
```diff
@@ -0,0 +1,266 @@
...
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: bootstrap-eks
+  name: capa-eks-bootstrap-manager-rolebinding
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:174]:
```suggestion
    {{- include "labels.common" . | nindent 4 }}
  name: {{ include "resource.default.name" . }}-manager
```

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 13]
```diff
@@ -0,0 +1,266 @@
...
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: capa-eks-bootstrap-manager-role
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:178]:
```suggestion
  name: {{ include "resource.default.name" . }}-manager
```

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 14]
```diff
@@ -0,0 +1,266 @@
...
+subjects:
+- kind: ServiceAccount
+  name: capa-eks-bootstrap-controller-manager
+  namespace: capa-eks-bootstrap-system
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:182]:
```suggestion
  name: {{ include "resource.default.name" . }}
  namespace: {{ include "resource.default.namespace" . }}
```

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 15]
```diff
@@ -0,0 +1,266 @@
...
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: bootstrap-eks
+  name: capa-eks-bootstrap-proxy-rolebinding
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:189]:
```suggestion
    {{- include "labels.common" . | nindent 4 }}
  name: {{ include "resource.default.name" . }}-proxy
```

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 16]
```diff
@@ -0,0 +1,266 @@
...
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: capa-eks-bootstrap-proxy-role
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:193]:
```suggestion
    {{- include "labels.common" . | nindent 4 }}
  name: {{ include "resource.default.name" . }}-proxy
```

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 17]
```diff
@@ -0,0 +1,266 @@
...
+subjects:
+- kind: ServiceAccount
+  name: capa-eks-bootstrap-controller-manager
+  namespace: capa-eks-bootstrap-system
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:197]:
```suggestion
  name: {{ include "resource.default.name" . }}
  namespace: {{ include "resource.default.namespace" . }}
```

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 18]
```diff
@@ -0,0 +1,266 @@
...
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: bootstrap-eks
+    control-plane: controller-manager
+  name: capa-eks-bootstrap-controller-manager-metrics-service
+  namespace: capa-eks-bootstrap-system
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:206]:
```suggestion
    {{- include "labels.common" . | nindent 4 }}
  name: {{ include "resource.default.name" . }}-metrics
  namespace: {{ include "resource.default.namespace" . }}
```

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 19]
```diff
@@ -0,0 +1,266 @@
...
+spec:
+  ports:
+  - name: https
+    port: 8443
+    targetPort: https
+  selector:
+    cluster.x-k8s.io/provider: bootstrap-eks
+    control-plane: controller-manager
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:214]:
```suggestion
    control-plane: controller-manager
    {{- include "labels.selector" . | nindent 4 }}
```

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 20]
```diff
@@ -0,0 +1,266 @@
...
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: bootstrap-eks
+    control-plane: controller-manager
+  name: {{ include "resource.default.name" . }}-bootstrap
+  namespace: giantswarm
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:223]:
```suggestion
    {{- include "labels.common" . | nindent 4 }}
  name: {{ include "resource.default.name" . }}-bootstrap
  namespace: {{ include "resource.default.namespace" . }}
```

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 21]
```diff
@@ -0,0 +1,266 @@
...
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      cluster.x-k8s.io/provider: bootstrap-eks
+      control-plane: controller-manager
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:229]:
```suggestion
      control-plane: controller-manager
      {{- include "labels.selector" . | nindent 6 }}
```

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 22]
```diff
@@ -0,0 +1,266 @@
...
+  template:
+    metadata:
+      labels:
+        cluster.x-k8s.io/provider: bootstrap-eks
+        control-plane: controller-manager
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:234]:
```suggestion
        control-plane: controller-manager
        {{- include "labels.common" . | nindent 8 }}
```

File: helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml [hunk 23]
```diff
@@ -0,0 +1,266 @@
...
+        - /manager
+        image: gcr.io/k8s-staging-cluster-api-aws/eks-bootstrap-controller:v0.6.4
+        imagePullPolicy: Always
+        name: manager
+      metadata:
+        annotations:
+          iam.amazonaws.com/role: {{ .Values.aws.role.arn }}:=""
+      securityContext:
+        fsGroup: 1000
+      serviceAccountName: capa-eks-bootstrap-controller-manager
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml:262]:
```suggestion
      serviceAccountName: {{ include "resource.default.name" . }}
```

**Review by TheoBrigitte — COMMENTED** [review]
helm helpers needs to be used, didn't go through the entire PR. Will continue later, or we can chat.

File: helm/cluster-api-provider-aws/templates/_helpers.tpl [hunk 24]
```diff
@@ -1,13 +1,3 @@
-# TEMPLATE-APP: This is set as a reasonable default, feel free to change.
```
[the remark this answers is not in the document]
Comment (anvddriesch) [reply helm/cluster-api-provider-aws/templates/_helpers.tpl:1]: The previous version was the default and it was replaced when I copypasted stuff from the other PR. It was not my intention. Thanks for the heads up!

**Comment by anvddriesch** [comment]
I decided to only include the infrastructure controller here and leave the other two controllers for a separate PR since it became to convoluted.

File: helm/cluster-api-provider-aws/templates/infrastructure-deployment.yaml [hunk 25]
```diff
@@ -0,0 +1,72 @@
...
+  selector:
+    matchLabels:
+      cluster.x-k8s.io/provider: infrastructure-aws
+      control-plane: capa-controller-manager
+      {{- include "labels.selector" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        cluster.x-k8s.io/provider: infrastructure-aws
+        control-plane: capa-controller-manager
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-aws/templates/infrastructure-deployment.yaml:21]:
```suggestion
        control-plane: capa-controller-manager
        {{- include "labels.selector" . | nindent 8 }}
```

**Review by TheoBrigitte — APPROVED** [review]

File: helm/cluster-api-provider-aws/templates/infrastructure-rbac.yaml [hunk 26]
```diff
@@ -0,0 +1,229 @@
...
+- apiGroups:
+  - controlplane.cluster.x-k8s.io
+  resources:
+  - awsmanagedcontrolplanes
+  - awsmanagedcontrolplanes/status
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
```
Comment (JosephSalisbury) [inline helm/cluster-api-provider-aws/templates/infrastructure-rbac.yaml:60]: is this repeated from https://github.com/giantswarm/cluster-api-provider-aws-app/pull/2/files#diff-355e6adc437980fc70d29b5a64aed16d7f326ee757ba89b8eb9436dc77454570R10?

File: helm/cluster-api-provider-aws/templates/infrastructure-deployment.yaml [hunk 27]
```diff
@@ -0,0 +1,73 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: infrastructure-aws
```
Comment (QuentinBisson) [inline helm/cluster-api-provider-aws/templates/infrastructure-deployment.yaml:5]: Can you add it into {{- include "labels.common" . | nindent 4 }} instead?

File: helm/cluster-api-provider-aws/templates/infrastructure-service.yaml [hunk 28]
```diff
@@ -0,0 +1,21 @@
...
+    {{- include "labels.common" . | nindent 4 }}
+  name: {{ include "resource.default.name" . }}-metrics
+  namespace: {{ include "resource.default.namespace" . }}
+spec:
+  ports:
+  - name: https
+    port: 8443
+    targetPort: https
+  selector:
+    cluster.x-k8s.io/provider: infrastructure-aws
```
Comment (QuentinBisson) [inline helm/cluster-api-provider-aws/templates/infrastructure-service.yaml:20]:
```suggestion
    cluster.x-k8s.io/provider: infrastructure-aws
    {{- include "labels.selector" . | nindent 4 }}
```

File: helm/cluster-api-provider-aws/templates/infrastructure-serviceaccount.yaml [hunk 29]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: infrastructure-aws
+    control-plane: controller-manager
```
Comment (QuentinBisson) [inline helm/cluster-api-provider-aws/templates/infrastructure-serviceaccount.yaml:6]:
```suggestion
    control-plane: controller-capa-manager
```

File: helm/cluster-api-provider-aws/Chart.yaml (hunk 2 above)
Comment (QuentinBisson) [inline helm/cluster-api-provider-aws/Chart.yaml:6]: Should we not set it?

File: helm/cluster-api-provider-aws/templates/infrastructure-deployment.yaml (hunk 27 above)
Comment (TheoBrigitte) [reply helm/cluster-api-provider-aws/templates/infrastructure-deployment.yaml]: Moved it into labels.common

File: helm/cluster-api-provider-aws/templates/infrastructure-service.yaml (hunk 28 above)
Comment (TheoBrigitte) [reply helm/cluster-api-provider-aws/templates/infrastructure-service.yaml]: fixed

File: helm/cluster-api-provider-aws/templates/infrastructure-serviceaccount.yaml (hunk 29 above)
Comment (TheoBrigitte) [reply helm/cluster-api-provider-aws/templates/infrastructure-serviceaccount.yaml]: fixed to `capa-controller-manager`

File: helm/cluster-api-provider-aws/Chart.yaml (hunk 2 above)
Comment (TheoBrigitte) [reply helm/cluster-api-provider-aws/Chart.yaml]: set to 0.6.4

**Review by MarcelMue — APPROVED** [review]
Fine with me - let's sort out the ARN stuff in a follow up

**Pull request merged**